src/x509.c

```diff
@@ -2012,31 +1958,58 @@ void* wolfSSL_X509V3_EXT_d2i(WOLFSSL_X509_EXTENSION* ext)
 
         /* subjectKeyIdentifier */
         case WC_NID_subject_key_identifier:
+        {
             WOLFSSL_MSG("subjectKeyIdentifier");
-            asn1String = wolfSSL_X509_EXTENSION_get_data(ext);
+
+            asn1String = wolfSSL_X509_EXTENSION_get_data_internal(ext);
             if (asn1String == NULL) {
                 WOLFSSL_MSG("X509_EXTENSION_get_data() failed");
                 return NULL;
             }
+
+            InitDecodedCert(&cert, NULL, 0, NULL);
+            ret = DecodeExtensionType((const byte*)asn1String->data,
+                asn1String->length, object->type, (byte)ext->crit, &cert, NULL);
+            if (ret != 0) {
+                WOLFSSL_MSG("DecodeExtensionType() failed");
+                return NULL;
+            }
+
             newString = wolfSSL_ASN1_STRING_new();
             if (newString == NULL) {
                 WOLFSSL_MSG("Failed to malloc ASN1_STRING");
                 return NULL;
             }
-            ret = wolfSSL_ASN1_STRING_set(newString, asn1String->data,
-                                                            asn1String->length);
+            ret = wolfSSL_ASN1_STRING_set(newString, cert.extSubjKeyId,
+                                          cert.extSubjKeyIdSz);
             if (ret != WOLFSSL_SUCCESS) {
                 WOLFSSL_MSG("ASN1_STRING_set() failed");
                 wolfSSL_ASN1_STRING_free(newString);
                 return NULL;
             };
             newString->type = asn1String->type;
             return newString;
+        }
 
         /* authorityKeyIdentifier */
         case WC_NID_authority_key_identifier:
+        {
             WOLFSSL_MSG("AuthorityKeyIdentifier");
 
+            asn1String = wolfSSL_X509_EXTENSION_get_data_internal(ext);
+            if (asn1String == NULL) {
+                WOLFSSL_MSG("X509_EXTENSION_get_data() failed");
+                return NULL;
+            }
+
+            InitDecodedCert(&cert, NULL, 0, NULL);
+            ret = DecodeExtensionType((const byte*)asn1String->data,
+                asn1String->length, object->type, (byte)ext->crit, &cert, NULL);
+            if (ret != 0) {
```

Comment:
Missing some edge cases to call `FreeDecodedCert`. Please make sure this is called as soon as cert is no longer needed and in all the error paths below.